### PR TITLE
add hw_7

### PR DESCRIPTION
**Pull request by KTsarenko**

### CODE REVIEW

File: homeworks/konstantin.tsarenko_KTsarenko/homework_7/hv_7.js [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+    const min = 50;
+    const max = 100;
+    return Math.floor(Math.random() * (max - min)) + min;
+}
+
+function getFibonacciSequence() {
+    const n = getRandomInt();
+    const fibonacci = [0, 1];
+
+    for (let i = 2; ; i++) {
```
Comment (markoffden) [inline homeworks/konstantin.tsarenko_KTsarenko/homework_7/hv_7.js]: why not limit your loop with max value you have already randomly generated?

File: homeworks/konstantin.tsarenko_KTsarenko/homework_7/hv_7.js [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+        const currentFib = fibonacci[i - 1] + fibonacci[i - 2];
+        if (currentFib < n) {
+            fibonacci[i] = currentFib;
+        } else {
+            break;
+        }
```
Comment (markoffden) [inline homeworks/konstantin.tsarenko_KTsarenko/homework_7/hv_7.js]: refactor your loop and there will be only one line of code here

File: homeworks/konstantin.tsarenko_KTsarenko/homework_7/getFibonacciSequence.test.js [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+    expect(fibonacci()[0]).toBe(0);
+});
+
+test('2 element of sequence to equal 1', () => {
+    expect(fibonacci()[1]).toBe(1);
+});
+
+describe('test sequence to be equal fibonacci', () => {
+    const fibonacciArr = fibonacci();
+    for (let i = 2; i < fibonacciArr.length - 3; i++) {
```
Comment (markoffden) [inline homeworks/konstantin.tsarenko_KTsarenko/homework_7/getFibonacciSequence.test.js]: just didn't get why `i < fibonacciArr.length - 3`? test doesn't need to check tree last element of the result array?

**Comment by markoffden** [comment]
files not in your HW directory should be removed from PR

File: homeworks/konstantin.tsarenko_KTsarenko/homework_7/getFibonacciSequence.test.js (hunk 3 above)
Comment (KTsarenko) [reply homeworks/konstantin.tsarenko_KTsarenko/homework_7/getFibonacciSequence.test.js]: this is my fail

File: homeworks/konstantin.tsarenko_KTsarenko/homework_7/hv_7.js (hunk 2 above)
Comment (KTsarenko) [reply homeworks/konstantin.tsarenko_KTsarenko/homework_7/hv_7.js]: update

**Comment by KTsarenko** [comment]
> files not in your HW directory should be removed from PR

Thought it because of what npm install did
need to revert the package-lock.json file?

File: homeworks/konstantin.tsarenko_KTsarenko/homework_7/hv_7.js [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+}
+
+function getFibonacciSequence() {
+    const n = getRandomInt();
+    const fibonacci = [0, 1];
+
+    for (let i = 2; i < n; i++) {
+        const currentFib = fibonacci[i - 1] + fibonacci[i - 2];
+        if (currentFib >= n) break;
+        fibonacci[i] = currentFib;
```
Comment (markoffden) [inline homeworks/konstantin.tsarenko_KTsarenko/homework_7/hv_7.js]: there is a `push()` method for this

**Comment by markoffden** [comment]
> > files not in your HW directory should be removed from PR
> 
> Thought it because of what npm install did
> need to revert the package-lock.json file?

yep, when you commit changes no files except your HW directory should be modified

File: homeworks/konstantin.tsarenko_KTsarenko/homework_7/hv_7.js (hunk 1 above)
Comment (markoffden) [reply homeworks/konstantin.tsarenko_KTsarenko/homework_7/hv_7.js]: sorry, just guided you wrong I guess... `while` loop could be more suitable here

**Comment by KTsarenko** [comment]
> yep, when you commit changes no files except your HW directory should be modified

returned the file to the original data

**Comment by markoffden** [comment]
cool, but you still have this package-lock.json in your PR

just make a commit from the same branch with deleted package-lock.json

**Comment by markoffden** [comment]
your branch has conflicts

you need to merge updated master branch into this branch^ resolve conflicts and update PR

**Comment by markoffden** [comment]
closed as outdated

**Pull request closed without merging**